elements/lisk-transactions/README.md

```diff
@@ -2,6 +2,91 @@
+
+Check out the Lisk SDK [Example Apps](https://github.com/LiskHQ/lisk-sdk-test-app) for simple [code examples of custom transaction types](https://github.com/LiskHQ/lisk-sdk-test-app/blob/development/hello_world/hello_transaction.js).
+
+### Lifecycle
+
+The lifecycle of a transaction in general in Lisk SDK can be summarized as follows:
+
+1. A transaction is created and signed (off-chain). The script to do it is in `src/create_and_sign.ts`.
+2. The transaction is sent to a network. This can be done by a third party tool (like `curl` or `Postman`), but also using Lisk Commander.
+   To send the default transactions for the protocol network, you can rely on the wallet Lisk UI tools - Lisk Hub and Mobile. All of the tools need to be authorized to access an HTTP API of a network node.
+3. A network node receives a transaction and after a lightweight schema validation, adds it to a transaction pool.
+4. In the transaction pool, the transactions are firstly `validated`. In this step, only static checks are performed. These include schema validation and signature validation.
+5. Validated transactions go to the `prepare` step defined in the transaction class, which to limit the I/O database operations prepares all the information relevant to properly `apply` or `undo` the transaction. The store with the prepared data is a parameter of the mentioned methods.
+6. Delegates are forging the valid transactions into the blocks and broadcasting the blocks to the network. Each network node performs the `apply` and `applyAsset` steps after the successful `validate` step.
+7. It is probable, especially shortly after a block is applied, that due to the decentralized network conditions a node does the `undo` step and the block containing all of the included transactions get reverted in favour of a competing block.
+
+While implementing a custom transaction, it is necessary to implement some of the mentioned steps. For most of them, a base transaction implements a default behaviour. As you feel more confident in using Lisk SDK, it is more likely for developers to override most of the base transaction methods, so the implementation is well-tailored and implemented with the best possible performance to the application's use case.
+
+### Interface
+
+Over the course of the alpha phase there will be significant changes in the Lisk protocol and implementation. We will be working towards the great developer experience while using Alpha SDK by reducing and optimizing the set of functions in custom transaction's API in the following releases. We strongly rely on the community feedback of how the interfaces can be improved.
+
+#### Required methods
+
+All of the abstract methods and properties on the base transaction's interface are required to implement. Those are:
+
+##### TYPE
+
+> static TYPE: number
```

Comment:
```suggestion
> static get TYPE (): number
```

Reply:
In Typescript is `static TYPE: number`.

Reply:
But aren't the custom transactions implemented in JS? At least they are in our examples.

Reply:
but still it's not a function.
In Javascript, it's equivalent to `Transaction.TYPE = 3`

Reply:
In JS you can also use static getters, what we do in our example apps, see:
https://github.com/LiskHQ/lisk-sdk-examples/blob/development/hello_world/hello_transaction.js#L8 and
https://github.com/LiskHQ/lisk-sdk-examples/blob/development/cashback/transactions/cashback_transaction.js#L9

Reply:
i dont think there is benefit of using getter here 🤔

Reply:
I dont either, but it's what we do in the examples ^^
And we also mention it here in section `Required methods`
In docs we changed this already and added a section for `Required properties` (see link above) - maybe that's better to understand?

Reply:
In ES5 you can assign a static var like `Transaction.TYPE = 3`, in ES6 you have an additional static field, but as all class properties in ES6 need to be implemented as a function, you could assign a static property as a getter function: `static get TYPE() { return 3; }`. 

I would be in favour of cutting the discussion here. We describe the interface of a TypeScript file, BaseTransaction, and we use TypeScript for that. We should describe our interfaces and leave the programming style up to individuals.